src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryTaskLauncher.java

```diff
@@ -119,7 +121,7 @@ public void destroy(String appName) {
 			.timeout(Duration.ofSeconds(this.deploymentProperties.getApiTimeout()))
 			.doOnSuccess(v -> logger.info("Successfully destroyed app {}", appName))
 			.doOnError(logError(String.format("Failed to destroy app %s", appName)))
-			.subscribe();
+			.block(Duration.ofSeconds(this.deploymentProperties.getApiTimeout()));
```

Comment:
We already have `timeout` configured for the same `API` timeout configuration and wouldn't this `timeout` be sufficient?

Reply:
this is what we do in other blocking calls, for example https://github.com/spring-cloud/spring-cloud-deployer-cloudfoundry/blob/744a1420521bb95f8ff249d254025879944074e7/src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryTaskLauncher.java#L108

Reply:
Yeah, to echo what @markpollack pointed out, I just did what the other blocking calls did.  If there is another approach I need to take, let me know.